gtsam/sfm/DsfTrackGenerator.h

```diff
@@ -0,0 +1,172 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file DsfTrackGenerator.h
+ * @date July 2022
+ * @author John Lambert
+ * @brief Identifies connected components in the keypoint matched graph.
+ */
+
+#pragma once
+#include <pybind11/stl.h>
+#include <Eigen/Core>
+
+#include <gtsam/base/DSFMap.h>
+#include <gtsam/geometry/Point2.h>
+
+#include <algorithm>
+#include <iostream>
+#include <map>
+#include <optional>
+#include <vector>
+
+namespace gtsam {
+
+typedef DSFMap<IndexPair> DSFMapIndexPair;
+typedef std::pair<size_t, size_t> ImagePair;
+typedef Eigen::MatrixX2i CorrespondenceIndices; // N x 2 array
+
+//struct Keypoints;
+using KeypointCoordinates = Eigen::MatrixX2d;
+
+
+struct Keypoints
+{
+  KeypointCoordinates coordinates;
+  // typedef'd for Eigen::VectorXd
+  boost::optional<gtsam::Vector> scales;
+  boost::optional<gtsam::Vector> responses;
+
+  Keypoints(const gtsam::KeypointCoordinates& coordinates): coordinates(coordinates) {}; // boost::none
+};
+
+using KeypointsList = std::vector<Keypoints>;
+using KeypointsVector = std::vector<Keypoints>; // TODO(johnwlambert): prefer KeypointsSet?
+using MatchIndicesMap = std::map<ImagePair, CorrespondenceIndices>;
+
+
+// @param camera index
+// @param 2d measurement
+// Implemented as named tuple, instead of std::pair (like SfmMeasurement in SfmTrack.h)
+struct NamedSfmMeasurement
```

Comment:
Does "Named" make a difference to the user? If not we could just call it SfmMeasurement.

Reply:
It would conflict with [the existing `SfmMeasurement`](https://github.com/borglab/gtsam/blob/develop/gtsam/sfm/SfmTrack.h#L32) here, and I really want the unambiguous named attributes, for parity with the Python side.

Reply:
why cant we use the existing tuple?